openbas-api/src/test/java/io/openbas/rest/InjectApiTest.java

```diff
@@ -1,175 +1,360 @@
-  }
-
-  @DisplayName("Retrieve inject for scenario")
-  @Test
-  @Order(3)
-  @WithMockObserverUser
-  void retrieveInjectForScenarioTest() throws Exception {
-    // -- EXECUTE --
-    String response = this.mvc
-        .perform(get(SCENARIO_URI + "/" + SCENARIO_ID + "/injects/" + INJECT_ID)
-            .accept(MediaType.APPLICATION_JSON))
-        .andExpect(status().is2xxSuccessful())
-        .andReturn()
-        .getResponse()
-        .getContentAsString();
-
-    // -- ASSERT --
-    assertNotNull(response);
-    assertEquals(INJECT_ID, JsonPath.read(response, "$.inject_id"));
-  }
-
-  @DisplayName("Add an inject for scenario")
-  @Test
-  @Order(4)
-  @WithMockPlannerUser
-  void updateInjectForScenarioTest() throws Exception {
-    // -- PREPARE --
-    Inject inject = this.injectRepository.findById(INJECT_ID).orElseThrow();
-    InjectInput input = new InjectInput();
-    String injectTitle = "A new title";
-    input.setTitle(injectTitle);
-    input.setInjectorContract(inject.getInjectorContract().getId());
-    input.setDependsDuration(inject.getDependsDuration());
-
-    // -- EXECUTE --
-    String response = this.mvc
-        .perform(put(SCENARIO_URI + "/" + SCENARIO_ID + "/injects/" + INJECT_ID)
-            .content(asJsonString(input))
-            .contentType(MediaType.APPLICATION_JSON)
-            .accept(MediaType.APPLICATION_JSON))
-        .andExpect(status().is2xxSuccessful())
-        .andReturn()
-        .getResponse()
-        .getContentAsString();
-
-    // -- ASSERT --
-    assertNotNull(response);
-    assertEquals(injectTitle, JsonPath.read(response, "$.inject_title"));
-  }
-
-  @DisplayName("Delete inject for scenario")
-  @Test
-  @Order(5)
-  @WithMockPlannerUser
-  void deleteInjectForScenarioTest() throws Exception {
-    // -- EXECUTE 1 ASSERT --
-    this.mvc.perform(delete(SCENARIO_URI + "/" + SCENARIO_ID + "/injects/" + INJECT_ID))
-        .andExpect(status().is2xxSuccessful());
-  }
+class InjectApiTest extends IntegrationTest {
+
+    static Exercise EXERCISE;
+    static Scenario SCENARIO;
+    static Document DOCUMENT1;
+    static Document DOCUMENT2;
+    static Team TEAM;
+    static String SCENARIO_INJECT_ID;
+
+    @Autowired
+    private MockMvc mvc;
+    @Autowired
+    private ScenarioService scenarioService;
+    @Autowired
+    private ExerciseService exerciseService;
+    @Autowired
+    private ExerciseRepository exerciseRepository;
+    @Autowired
+    private ScenarioRepository scenarioRepository;
+    @Autowired
+    private InjectRepository injectRepository;
+    @Autowired
+    private DocumentRepository documentRepository;
+    @Autowired
+    private CommunicationRepository communicationRepository;
+    @Autowired
+    private InjectExpectationRepository injectExpectationRepository;
+    @Autowired
+    private TeamRepository teamRepository;
+    @Autowired
+    private InjectorContractRepository injectorContractRepository;
+
+    @BeforeAll
+    void beforeAll() {
```

Comment:
I might miss something but wouldn't it be better to mock the repositories instead of creating and deleting elements in a @BeforeAll and @AfterAll annotation ?

Reply:
In this case, it is an integration test, so I usually don't use mocks (just for 3rd services)... but, it's true that I'd prefer another way to load and clean data for the tests, perhaps a sql script...

Reply:
Ok, I usually mock DB calls and 3rd party services in my integration tests but I understand why you'd want to keep them. Not for this PR but maybe we could think of a throwable DB used only for tests.